### PR TITLE
Properly parse ID3v2.2 tags

### PR DESCRIPTION
**Pull request by moffatman**
Also add some explicit type annotations

This solves #1 (I tested the file here: https://github.com/sanket143/id3/pull/6#issuecomment-890592379)

### CODE REVIEW

File: lib/id3.dart [hunk 1]
```diff
@@ -304,19 +319,24 @@ class MP3Instance {
 }
 
 int parseSize(List<int> block, int major_v) {
-  assert(block.length == 4);
-
   int len;
   if (major_v == 4) {
+    assert(block.length == 4);
```
Comment (sanket143) [inline lib/id3.dart:324]: I think we shouldn't be using assert statements in production. They are only useful during development and testing.

Comment (moffatman) [reply lib/id3.dart:324]: It doesn't run in production. That's why I didn't know it was broken. So it's OK to keep it in my opinion, it's self-documenting.

**Comment by sanket143** [comment]
Hello @moffatman, would you be interested in being a collaborator?

**Comment by sanket143** [comment]
@StarBattle08 published

**Pull request merged**